File: src/BuildingBlocks/Remote.Shell.Interrupt.BuildingBlocks/Exceptions/ApiErrorResponse.cs
namespace Remote.Shell.Interrupt.BuildingBlocks.Exceptions;

public class ApiErrorResponse
{
    public string Title { get; set; } = string.Empty;
    public int Status { get; set; }
    public string Detail { get; set; } = string.Empty;
    public IReadOnlyDictionary<string, string[]>? Errors { get; set; }
}

File: client/src/lib/types/Common/ApiErrorResponse.ts
/**
 * Shape of the JSON error payload returned by the API's
 * ExceptionHandlingMiddleware.
 *
 * The backend serializes this payload with a plain
 * `JsonSerializer.Serialize` call (not through MVC), so property names
 * keep their PascalCase declaration names — unlike regular DTO bodies,
 * which are camelCase.
 */
export type ApiErrorResponse = {
  Status: number;
  Title: string;
  Detail: string;
  Errors?: { [key: string]: string[] };
};
